src/lib/urn.ts

```diff
@@ -121,14 +121,14 @@ export function isThirdParty(urn?: string) {
   return decodedURN.type === URNType.COLLECTIONS_THIRDPARTY
 }
 
-export function decodeURN(urn: URN): DecodedURN {
+export function decodeURN<T extends URNType>(urn: URN): DecodedURN<T> {
```

Comment:
Although it might get a bit confusing to use, `decodeURN` works without generics due to it being able to parse any kind of URN, that is, if you type it beforehand, your type might be wrong.
The correct way of using it is to decode the URN and then use an if clause on the type so, if the if clause is true about the type, then the block surrounding it will type the URN as the type it is.
For example (taken from `CreateSingleItemModal.tsx:255`):
```
       let decodedCollectionUrn: DecodedURN<any> | null = collection?.urn ? decodeURN(collection.urn) : null
        let urn: string | undefined
        if (
          decodedCollectionUrn &&
          decodedCollectionUrn.type === URNType.COLLECTIONS_THIRDPARTY &&
          decodedCollectionUrn.thirdPartyCollectionId
        ) {
          urn = buildThirdPartyURN(decodedCollectionUrn.thirdPartyName, decodedCollectionUrn.thirdPartyCollectionId, uuid.v4())
        }
```

Reply:
awesome! thanks for the explanation 💯 , updated!